lib/dialects/abstract/connection-manager.js

```diff
@@ -85,20 +90,7 @@ class ConnectionManager {
 
     if (!config.replication) {
       this.pool = Pooling.createPool({
-        create: () => new Promise(resolve => {
-          this
-            ._connect(config)
-            .tap(() => {
-              this.poolError = null;
-            })
-            .then(resolve)
-            .catch(e => {
-              // dont throw otherwise pool will release _dispense call
-              // which will call _connect even if error is fatal
-              // https://github.com/coopernurse/node-pool/issues/161
-              this.poolError = e;
-            });
-        }),
+        create: () => this._connect(config),
         destroy: connection => {
```

Comment:
suggest changing name <code>connection</code> to <code>mayBeConnection</code>

And, I think "connection destroy" shouldn't be logged if error.

```javascript
if (mayBeConnection instanceof Error) {
  return Promise.resolve(); // avoid "connection destroy" debug log
}
return this._disconnect(mayBeConnection).tap(() => {
  debug('connection destroy');
});
```